### ISSUE TITLE
Shorten and linkify commit hashes

### ISSUE THREAD

**Issue by j-f1** [GitHawkApp/GitHawk#2313]
Example comment: https://github.com/MichaelMure/git-bug/issues/69#issuecomment-430811694

On GitHub.com, the hash in the comment is converted to `code`, truncated to a smaller length, and linked to the actual commit.

Can GitHawk do this too?

<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.23.0 (1539742783)
Device: iPhone 8 (iOS 12.0.1)
TestFlight: true
</pre>
</details>

**Comment by rnystrom** [comment]
Is there a regex rule for commit hashes?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by j-f1** [comment]
Yes, but unfortunately it’s `[\da-f]{7,40}`. You’d then have to check matching strings to see if they correspond to a real commit (on the plus side, if a commit is found, the SHA can be cached indefinitely). As a start, truncating and linkifying the 40-character ones would make them a lot easier to read.

**Comment by j-f1** [comment]
Aha: GitHub has [an API](https://developer.github.com/v3/markdown/) for rendering Markdown.

**Comment by j-f1** [comment]
There’s also a way to get GitHub to send you the HTML for a given comment.

**Comment by j-f1** [comment]
Could you use NSAttributedString to convert the HTML to a native string?

**Comment by rnystrom** [comment]
That uses a WebKit underneath and has to be done on the main thread.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by tooomm** [comment]
At least the truncated display (first 7 digits) would help to view posts where commits are referenced...
